### PR TITLE
[RFC] Topology for synchronize playback

### PR DESCRIPTION
**Pull request by bardliao**
The PR is to implement a topology for synchronize playback. We use demux to duplicate audio data for two DAIs.

### CODE REVIEW

File: tools/topology/m4/dai.m4 [hunk 1]
```diff
@@ -200,4 +200,9 @@ define(`DAI_ADD',
 `DEBUG_DAI($3, $4)'
 )
 
+define(`DAI_ADD_SCHED',
```
Comment (lgirdwood) [inline tools/topology/m4/dai.m4]: Can you add some comments here on usage and what its for.

File: tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4 [hunk 2]
```diff
@@ -185,9 +195,6 @@ DAI_CONFIG(ALH, 3, 1, SDW0-Capture)
 #ALH SDW1 Pin2 (ID: 2)
 DAI_CONFIG(ALH, 0x102, 2, SDW1-Playback)
 
-#ALH SDW2 Pin2 (ID: 3)
-ifdef(`MONO', `', `DAI_CONFIG(ALH, 0x202, 3, SDW2-Playback)')
```
Comment (lgirdwood) [inline tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4]: These two changes for SDW2 look like they are fixing another issues - separate patches ?

File: tools/topology/sof/pipe-simple-playback.m4 [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Low Latency Passthrough Pipeline and PCM
+#
+# Pipeline Endpoints for connection are :-
+#
+#  B0 --> sink DAI0
```
Comment (lgirdwood) [inline tools/topology/sof/pipe-simple-playback.m4]: can we rename this to pipe-dai-playback.m4

File: tools/topology/sof/pipe-dai-sched-playback.m4 [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# DAI Playback connector
+
```
Comment (lgirdwood) [inline tools/topology/sof/pipe-dai-sched-playback.m4]: Please add some comments here to describe the topology.

File: tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4 (hunk 2 above)
Comment (bardliao) [reply tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4]: > These two changes for SDW2 look like they are fixing another issues - separate patches ?

Yes, done

File: tools/topology/sof/pipe-simple-playback.m4 (hunk 3 above)
Comment (bardliao) [reply tools/topology/sof/pipe-simple-playback.m4]: > can we rename this to pipe-dai-playback.m4

There is another pipe-dai-playback.m4 in the same folder already.

File: tools/topology/sof/pipe-dai-sched-playback.m4 (hunk 4 above)
Comment (bardliao) [reply tools/topology/sof/pipe-dai-sched-playback.m4]: > Please add some comments here to describe the topology.

Done

File: tools/topology/sof/pipe-simple-playback.m4 (hunk 3 above)
Comment (lgirdwood) [reply tools/topology/sof/pipe-simple-playback.m4]: pipe-dai-endpoint

File: tools/topology/sof/pipe-simple-playback.m4 [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Low Latency Passthrough Pipeline and PCM
```
Comment (ranj063) [inline tools/topology/sof/pipe-simple-playback.m4]: this comment is wrong

Comment (ranj063) [reply tools/topology/sof/pipe-simple-playback.m4]: besides this is hardly a pipeline with the SectionGraph defined elsewhere. All you have is a buffer?

File: tools/topology/sof/pipe-dai-sched-playback.m4 [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+#
+W_PIPELINE(SCHED_COMP, SCHEDULE_PERIOD, SCHEDULE_PRIORITY, SCHEDULE_CORE, SCHEDULE_TIME_DOMAIN, pipe_dai_schedule_plat)
+
+#
+# Graph connections to pipelines
+
+P_GRAPH(DAI_NAME, PIPELINE_ID,
+	LIST(`		', `dapm(N_DAI_OUT, DAI_BUF)'))
+
+indir(`define', concat(`PIPELINE_PLAYBACK_SCHED_COMP_', PIPELINE_ID), N_DAI_OUT)
```
Comment (ranj063) [inline tools/topology/sof/pipe-dai-sched-playback.m4]: is this needed if you're not using the N_DAI_OUT here?

File: tools/topology/sof/pipe-simple-playback.m4 (hunk 5 above)
Comment (bardliao) [reply tools/topology/sof/pipe-simple-playback.m4]: > besides this is hardly a pipeline with the SectionGraph defined elsewhere. All you have is a buffer?

I will have a buffer with a pipeline id which is required for connecting to a demux.

File: tools/topology/sof/pipe-dai-sched-playback.m4 (hunk 6 above)
Comment (bardliao) [reply tools/topology/sof/pipe-dai-sched-playback.m4]: > is this needed if you're not using the N_DAI_OUT here?

Yeah, we don't need it. Thanks @ranj063

**Pull request merged**